src/Mod/Sketcher/Gui/SnapManager.h

```diff
@@ -1,134 +1,134 @@
-    double lastMouseAngle;
-
-    double snapAngle;
-
-    /// Observer to track all the needed parameters.
-    std::unique_ptr<SnapManager::ParameterObserver> pObserver;
-};
-
-
-}  // namespace SketcherGui
-
-
-#endif  // SKETCHERGUI_SnapManager_H
+/***************************************************************************
+ *   Copyright (c) 2023 Pierre-Louis Boyer <pierrelouis.boyer@gmail.com>   *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ *   SnapManager initially funded by the Open Toolchain Foundation         *
+ ***************************************************************************/
+
+#ifndef SKETCHERGUI_SnapManager_H
+#define SKETCHERGUI_SnapManager_H
+
+#include <App/Application.h>
+
+#include <Base/Tools2D.h>
+#include <Base/Vector3D.h>
+
+namespace Part
+{
+class GeomLineSegment;
+class GeomArcOfCircle;
+}  // namespace Part
+
+namespace SketcherGui
+{
+
+class ViewProviderSketch;
+
+
+class ViewProviderSketchSnapAttorney
+{
+private:
+    static inline int getPreselectPoint(const ViewProviderSketch& vp);
+    static inline int getPreselectCross(const ViewProviderSketch& vp);
+    static inline int getPreselectCurve(const ViewProviderSketch& vp);
+
+    friend class SnapManager;
+};
+
+/* This class is used to manage the overriding of mouse pointer coordinates in Sketcher
+ *  (in Edit-Mode) depending on the situation. Those situations are in priority order :
+ *  1 - Snap at angle: For tools like Slot, Arc, Line, Ellipse, this enables to constrain the angle
+ * at steps of 5° (or customized angle). This is useful to make features at a certain angle (45° for
```

Comment:
This probably should be fixed to be simply degrees

Reply:
I don't see any issue with using non-ASCII characters (especially in comments) -- many people's names include non-ASCII characters, for example, so a proper copyright attribution should allow for that. Fixing the encoding here should address any problem with the characters.

Reply:
![image](https://github.com/FreeCAD/FreeCAD/assets/747404/de14520b-d7e4-4ee7-86c7-a18e33bc5506)

This is how it is rendered on github for me - that's why I assumed that it is saved with incorrect character. But I see that on preview it looks fine - strange.